engines/sherlock/surface.h

```diff
@@ -0,0 +1,62 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#ifndef SHERLOCK_GRAPHICS_H
+#define SHERLOCK_GRAPHICS_H
+
+#include "common/rect.h"
+#include "graphics/surface.h"
+#include "sherlock/resources.h"
+
+namespace Sherlock {
+
+class Surface : public Graphics::Surface {
+private:
+	bool _freePixels;
+
+	bool clip(Common::Rect &srcBounds, Common::Rect &destBounds);
+protected:
+	virtual void addDirtyRect(const Common::Rect &r) {}
+public:
+	Surface(uint16 width, uint16 height);
+	Surface();
+	virtual ~Surface();
```

Comment:
Another good reason why publicly inherting from `Graphics::Surface` might be a bad idea: It allows you to keep pointers of type `Graphics::Surface *` to a Sherlock::Surface object around. On `delete` through that pointer only `Graphics::Surface::~Surface` is executed.